src/ResourceManager/Profile/Commands.Profile/Models/RMProfileClient.cs

```diff
@@ -121,8 +121,16 @@ public RMProfileClient(AzureRMProfile profile)
                         newTenant == null &&                         
                         TryGetTenantSubscription(token, account, environment, tenant, subscriptionId, subscriptionName, out tempSubscription, out tempTenant))
                     {
-                        newTenant = tempTenant;
-                        newSubscription = tempSubscription;
+                        if (tempSubscription == null && i + 1 < tenants.Count())
```

Comment:
BTW, this will change the logic of selecting the tenants. Now if tenant does not have any subscription - Last tenant will be selected. Somewhere in the code we say that "First subscription will be selected"